packages/aws-cdk-lib/verify-stripped-exp.js

```diff
@@ -0,0 +1,134 @@
+// +------------------------------------------------------------------------------------------------
+// | this script runs post pack to verify that experimental modules in aws-cdk-lib includes **only**  L1 autogenerated files. 
+// | The purpose is to avoid publishing L2 of experimental modules with aws-cdk-lib
+// |
+const { spawnSync } = require('child_process');
+const os = require('os');
+const fs = require('fs-extra');
+const path = require('path');
+
+
+const JS_DIST_LIB = 'dist/js/';
+
+async function main(tempDir) {
+  const cwd = process.cwd();
+  const awsCdkModulesRepoPath = path.join(findWorkspacePath(),'packages', '@aws-cdk');
+  process.chdir(tempDir);
+  const version = require('./package.json').version;
+  const tarFullPath = path.join(cwd, JS_DIST_LIB, `aws-cdk-lib-${version}.tgz`);
+
+  // install the tarball in a temp directory
+  exec('npm', ['init', '-y']);
+  exec('npm', ['install', tarFullPath]);
+  const installedAwsCdkLibPath = path.join('node_modules', 'aws-cdk-lib', 'lib');
+
+  for (const module of fs.readdirSync(awsCdkModulesRepoPath)) {
+    const pkgJson = require(path.join(awsCdkModulesRepoPath, module, 'package.json'));
+    if (pkgJson['stability'] !== 'experimental') {
+      continue;
+    }
+    if (pkgJson['cdk-build'].cloudformation) {
+      // if a cfn module, verify only the allowed files exists
+      const files = await getAllFiles(path.join(installedAwsCdkLibPath, module));
+      files.forEach(file => {
+        if (!isAllowedFile(file)) {
+          throw `only L1 auto generated files are allowed in experimental modules included in aws-cdk-lib, the file ${file} exists in ${module}`;
+        }
+      });
+    } else {
+      // not a cfn module, verify it was entirely removed
+      if (fs.existsSync(path.join(installedAwsCdkLibPath, module))) {
+        throw `only L1s of experimental modules are allowed in aws-cdk-lib. The higher level module ${module} was included`;
+      }
+    }
+  }
+}
+
+const tempDir = fs.mkdtempSync(os.tmpdir());
+
+main(tempDir).then(
+  () => {
+    fs.removeSync(tempDir);
+    process.exit(0);
+  },
+  (err) => {
+    process.stderr.write(`${err.toString()}\n`);
+    process.stderr.write(`did not delete the temp folder: ${tempDir}`)
+    process.exit(1);
+  },
+);
+
+
+/**
+ * Spawn sync with error handling
+ */
+function exec(cmd, args) {
+  const proc = spawnSync(cmd, args);
+
+  if (proc.error) {
+    throw proc.error;
+  }
+
+  if (proc.status !== 0) {
+    if (proc.stdout || proc.stderr) {
+      throw new Error(`[Status ${proc.status}] stdout: ${proc.stdout?.toString().trim()}\n\n\nstderr: ${proc.stderr?.toString().trim()}`);
+    }
+    throw new Error(`${cmd} exited with status ${proc.status}`);
+  }
+
+  return proc;
+}
+
+const GENERATED_SUFFIX_REGEX = new RegExp(/generated\.(js|ts|d\.ts)$/);
+const ALLOWED_FILES = ['.jsiirc.json', 'index.ts', 'index.js', 'index.d.ts'];
```

Comment:
`.ts` files shouldn't be published so we can remove it from the allowed so thats enforced as well.